drivers/applications/drv8825_demo.cpp

```diff
@@ -0,0 +1,37 @@
+#include <libhal-util/serial.hpp>
+#include <libhal-util/steady_clock.hpp>
+#include <libhal/units.hpp>
+
+#include "../include/drv8825.hpp"
+#include "../hardware_map.hpp"
+
+using namespace hal::literals;
+using namespace std::chrono_literals;
+namespace sjsu::drivers {
+
+void application(application_framework& p_framework)
+{
+  // configure drivers
+  auto& clock = *p_framework.steady_clock;
+  auto& terminal = *p_framework.terminal;
+  
+  auto m_drv8825 = drv8825(
+            *p_framework.out_pin3, 
+            *p_framework.out_pin4, 
+            *p_framework.steady_clock, 
+            drv8825::step_factor::one, 
+            2048, 
+            450us,
+            {p_framework.out_pin0,p_framework.out_pin1,p_framework.out_pin2}
+        );
+  
+    hal::print<64>(terminal, "starting motor\n");
```

Comment:
```suggestion
    hal::print(terminal, "starting motor\n");
```

You can print string literals directly without needing an extra stack buffer so that parameter can be eliminated.